src/components/TransactionDialog/TransactionDialog.tsx

```diff
@@ -15,58 +35,183 @@ export type TransactionDialogProps = Pick<ActionDialogProps, 'className'> & {
-    tooltip: '',
+    tooltip: 'Processing',
+    animation: {
+      data: firstStepAnimation,
+      size: {
+        width: 216,
+        height: 216,
+      },
+      loop: true,
+    },
   },
   [ExtrinsicStatus.Unsigned]: {
-    title: 'Waiting for signature...',
+    title: 'Sign in Polkadot',
     description: 'Please sign the transaction using the Polkadot browser extension.',
     tooltip: 'Signature',
+    animation: {
+      data: secondStepAnimation,
+      size: {
+        width: 288,
+        height: 216,
+      },
+      loop: false,
+    },
   },
   [ExtrinsicStatus.Signed]: {
-    title: 'Waiting for confirmation...',
+    title: 'Processing transaction',
     description:
       'Your transaction has been signed and sent. Please wait for the blockchain confirmation. This should take about 15 seconds.',
     tooltip: 'Confirmation',
+    animation: {
+      data: thirdStepAnimation,
+      size: {
+        width: 216,
+        height: 216,
+      },
+      loop: true,
+    },
   },
   [ExtrinsicStatus.Syncing]: {
-    title: 'Waiting for data propagation...',
+    title: 'Propagating changes',
     description:
       "Your transaction has been accepted and included into the blockchain. Please wait till it's picked up by the indexing node. This should take up to 15 seconds.",
     tooltip: 'Propagation',
+    animation: {
+      data: fourthStepAnimation,
+      size: {
+        width: 144,
+        height: 144,
+      },
+      loop: true,
+    },
+  },
+  [ExtrinsicStatus.Error]: {
+    title: 'Something went wrong...',
+    description:
+      'An unexpected error was encountered. If this persists, our Discord community may be a good place to find some help.',
+    tooltip: 'Propagation',
+    animation: {
+      data: errorAnimation,
+      size: {
+        width: 288,
+        height: 264,
+      },
+      loop: false,
+    },
   },
 }
 
 export const TransactionDialog: React.FC<TransactionDialogProps> = ({ status, onClose, ...actionDialogProps }) => {
+  const [polkadotLogoVisible, setPolkadotLogoVisible] = useState(false)
+  const initialStatusRef = useRef<number | null>(null)
+  const nonUploadTransaction = initialStatusRef.current === ExtrinsicStatus.Unsigned
+  const error = status === ExtrinsicStatus.Error
   const stepDetails =
-    status != null && status !== ExtrinsicStatus.Error && status !== ExtrinsicStatus.Completed
-      ? TRANSACTION_STEPS_DETAILS[status]
+    status != null
+      ? TRANSACTION_STEPS_DETAILS[status === ExtrinsicStatus.Completed ? ExtrinsicStatus.Syncing : status]
       : null
 
-  const canCancel = status === ExtrinsicStatus.ProcessingAssets || status === ExtrinsicStatus.Unsigned
+  useEffect(() => {
+    if (status !== null && initialStatusRef.current === null) {
+      initialStatusRef.current = status
+    }
+    if (status === null) {
+      initialStatusRef.current = null
+    }
```

Comment:
Why do we need this `initialStatusRef`?

Reply:
This is hack to determine whether it's upload transaction or something else, like delete for example. If it's upload transaction, then `initialStatusRef.current === 0`

Reply:
Why do we need this information?

Reply:
Because we need to know how many steps we have (we show number of steps at the top of dialog). Non upload transaction has 1 step less than upload transaction.